File: akka-benchmark/src/main/java/io/sarl/akka/eventbus/CustomEventBus.java
package io.sarl.akka.eventbus;

import akka.actor.ActorRef;
import akka.event.japi.LookupEventBus;

public class CustomEventBus extends LookupEventBus<Message, ActorRef, String> {

    @Override
    public int mapSize() {
        return 128;
    }

    @Override
    public int compareSubscribers(ActorRef a, ActorRef b) {
        return a.compareTo(b);
    }

    @Override
    public String classify(Message event) {
        return event.getChannel();
    }

    @Override
    public void publish(Message event, ActorRef subscriber) {
        subscriber.tell(event, ActorRef.noSender());
    }
}

File: akka-benchmark/src/main/kotlin/io/sarl/akka/eventbus/lookup/CustomEventBus.kt
package io.sarl.akka.eventbus.lookup

import akka.actor.ActorRef
import akka.event.japi.LookupEventBus

class CustomEventBus : LookupEventBus<Message, ActorRef, String>() {

    override fun mapSize(): Int {
        return 128
    }

    override fun compareSubscribers(a: ActorRef, b: ActorRef): Int {
        return a.compareTo(b)
    }

    override fun classify(event: Message): String {
        return event.channel
    }

    override fun publish(event: Message, subscriber: ActorRef) {
        subscriber.tell(event, ActorRef.noSender())
    }
}
